drivers/SmartThings/zwave-switch/src/qubino-switches/qubino-relays/qubino-flush-2-relay/init.lua

```diff
@@ -11,43 +11,163 @@
 -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
+local st_device = require "st.device"
+local capabilities = require "st.capabilities"
+--- @type st.zwave.CommandClass
+local cc = require "st.zwave.CommandClass"
+--- @type st.zwave.CommandClass.Basic
+local Basic = (require "st.zwave.CommandClass.Basic")({ version = 1 })
+--- @type st.zwave.CommandClass.SwitchBinary
+local SwitchBinary = (require "st.zwave.CommandClass.SwitchBinary")({ version = 1 })
+--- @type st.zwave.CommandClass.Meter
+local Meter = (require "st.zwave.CommandClass.Meter")({ version = 3 })
+--- @type st.zwave.CommandClass.SensorMultilevel
+local SensorMultilevel = (require "st.zwave.CommandClass.SensorMultilevel")({ version = 7 })
+--- @type st.zwave.constants
+local constants = require "st.zwave.constants"
 
-local QUBINO_FLUSH_2_RELAY_FINGERPRINT = {mfr = 0x0159, prod = 0x0002, model = 0x0051}
+local QUBINO_FLUSH_2_RELAY_FINGERPRINT = { mfr = 0x0159, prod = 0x0002, model = 0x0051 }
 
 local function can_handle_qubino_flush_2_relay(opts, driver, device, ...)
   return device:id_match(QUBINO_FLUSH_2_RELAY_FINGERPRINT.mfr, QUBINO_FLUSH_2_RELAY_FINGERPRINT.prod, QUBINO_FLUSH_2_RELAY_FINGERPRINT.model)
 end
 
 local function component_to_endpoint(device, component_id)
-    if component_id == "main" then
-      return { 1 }
-    elseif component_id == "extraTemperatureSensor" then
-      return { 3 }
-    else
-      local ep_num = math.floor(component_id:match("switch(%d)"))
-      return { ep_num and tonumber(ep_num)}
+  return { 1 }
+end
+
+local function find_child(parent, src_channel)
+  if src_channel == 1 then
+    return parent
```

Comment:
you should just return `nil` if there isn't a child.

Reply:
if I return nil in this place reports for ep `{1}` are not handled

Reply:
This can be done by setting the `component_to_endpoint` function, but I'm curious if we've had to do this for other Z-Wave devices, looking at the code it looks like we have always defaulted everything to ep 0, but it looks like this device doesn't use 0 at all?

Reply:
ep 0 is the same as unencapsulated, I'm pretty sure

Reply:
I mentioned in one of comments that ep 0 in the device is to report aggregated information about it's state for example in one of switches is on then report will return ON_ENABLE value, also power/energy meter reports for is sum of values in ep 1 and 2. Ealier I did the driver which includes parent (ep 0) child_switch_1 (ep 1) child_switch_2 (ep 2) optional_child_temp_sensor (ep 3), but I've changed it to parent (ep 1) child_switch_2 (ep 2) optional_child_temp_sensor (ep 3) to provide identical behavior to DTH to make it pass migration tests.

Reply:
And not exactly it can be done by conponent_to_endpoint
```
function ZwaveDevice:emit_event_for_endpoint(endpoint, event)
  local find_child_fn = self:get_field(base_device.FIND_CHILD_KEY)
  if find_child_fn ~= nil then
    local child = find_child_fn(self, endpoint)
    if child ~= nil then
      child:emit_event(event)
      return
    end
  else
    local component_id = self:endpoint_to_component(endpoint)
    self:emit_component_event(self.profile.components[component_id], event)
  end
end
```
I need to use emit_event_for_endpoint which does not use the component_to_endpoint function and as you can see here parent device has find_child function so it will not use any of comp -> ep or ep -> comp functions

Reply:
Yeah, this is a bug, so you are right for now you need to return the parent from the find_child_fn.  I can create a PR to update this in the future, that won't break returning the parent, but right now the way it works is different between how zigbee and zwave works